git/intermediate_git/a_deeper_look_at_git.md

```diff
@@ -114,7 +114,49 @@ Let's start by looking a bit closer at what happened here. When you ran `git res
 
 Now let's say we want to move where HEAD points to but *don't* want to touch the staging area. If we want to leave the index alone, you can use `git reset --soft`. This would only perform the first part of `git reset` where the HEAD is moved to point somewhere else.
 
-The last part of reset we want to touch upon is `git reset --hard`. What this does is it performs all the steps of `git reset`, moving the HEAD and updating the index, but it *also* updates the working directory. This is important to note because it can be dangerous as it can potentially destroy data. A hard reset overwrites the files in the working directory to make it look exactly like the staging area of wherever HEAD ends up pointing to. Similarly to `git commit --amend`, a hard reset is a destructive command which overwrites history. This doesn't mean you should completely avoid it if working with shared repositories on a team with other developers. You should, however, **make sure you know exactly why you're using it, and that your coworkers are also aware of how and why you're using it.**
+If we want to use the index for `git reset`, we have to first determine where the current branch(`HEAD`)is pointing. So, if we're on branch `master` with this series of commits:
+
+~~~bash
+- A - B - C (master)
+~~~
+
+The `HEAD` points to `C` and the index matches `C`.
+
+When we run `git reset --soft B`, `master`(and thus `HEAD`) now points to `B`, but the index still has
```

Comment:
This looks the exact same as the Stackoverflow answer? we can't use this, it's plagiarism.

But I don't think we need a deep technical explanation like this anyway. The main point in the issue was our content makes it hard for learners to understand when using `git reset --soft` is useful. 

Providing learners with one or two practical problems they may encounter and how they can use `git reset --soft` to solve those problems would be the best way of teaching this.

Something similar to this [this answer](https://stackoverflow.com/a/59675191) would be perfect.

Reply:
Sorry, I will fix this and do what you asked.